File: backend/app/schemas/shpi_map.py
from pydantic import BaseModel


class ShpiMapCodeItem(BaseModel):
    code: str
    region_code: str
    current_value: int
    status: str


class ShpiMapRegionItem(BaseModel):
    code: str
    name: str


class ShpiMapResponse(BaseModel):
    regions: list[ShpiMapRegionItem]
    region_codes: list[str]
    codes: list[str]
    cells: list[ShpiMapCodeItem]

File: frontend/src/api/shpiMap.ts
import { apiFetch } from './client';

export type ShpiMapStatus = 'green' | 'gray' | 'red';

export interface ShpiMapCodeItem {
  code: string;
  region_code: string;
  current_value: number;
  status: ShpiMapStatus;
}

export interface ShpiMapRegionItem {
  code: string;
  name: string;
}

export interface ShpiMapResponse {
  regions: ShpiMapRegionItem[];
  region_codes: string[];
  codes: string[];
  cells: ShpiMapCodeItem[];
}

export interface OfficialShpiCounterItem {
  package_type: string;
  region_code: string;
  current_value: number;
  used_count: number;
  last_used_date: string | null;
}

export const getShpiMap = (): Promise<ShpiMapResponse> =>
  apiFetch<ShpiMapResponse>('/admin/shpi-map');

export const getOfficialShpiCounters = (fresh = false): Promise<OfficialShpiCounterItem[]> =>
  apiFetch<OfficialShpiCounterItem[]>(`/admin/official-shpi/counters${fresh ? '?fresh=true' : ''}`);
